source/binding/Expression.cpp

```diff
@@ -165,7 +165,10 @@ const Expression& Expression::bindLValue(const ExpressionSyntax& lhs, const Type
 const Expression& Expression::bindRValue(const Type& lhs, const ExpressionSyntax& rhs,
                                          SourceLocation location, const BindContext& context) {
     Compilation& comp = context.scope.getCompilation();
-    Expression& expr = create(comp, rhs, context, BindFlags::None, &lhs);
+    bitmask<BindFlags> extraFlags = context.instance || (context.flags & BindFlags::EnumInitializer)
```

Comment:
Is the streaming operator actually disallowed for enum initializers or ports? Actually, interestingly enough, it seems like streaming operators might not actually be allowed in constant expressions according to the grammar, but that seems like an oversight in the grammar.

Reply:
This is not precisely defined in the standard.
1. I currently allow streaming operator in parameter initializers. The standard says "assignment" not "assignment-like context". If it were "assignment-like context", constant expression to initialize parameters should be allowed. It depends on interpretation. Let me know your decision. I can change it.
2. I currently disallow it for ports. Again, if it is "assignment-like context", ports should be allowed. The complexity is for instance array, where the type is needed to divide the expression among  instances, and there is no well-defined type for streaming operators.
3. I currently disallow it for enum initializers. There are special code and I have not paid attention to it. When it becomes important to support it, it can be a TODO item.

Reply:
For constant expressions, [11.2.1] page 255 says

> Constant expressions can use any of the operators defined in Table 11-1.

and streaming operators are in Table 11-1.

Reply:
All the simulators I tried allow it for ports, so I think it makes sense to allow it. We can explicitly disallow it for dividing among instance array ports, since that doesn't seem to be supported by other tools (and it's confusing anyway). Also, if we allow it for parameters we should allow it for enum initializers as well.

Reply:
It is good to know this information, and streaming operators should be supported in port connection and enum initializer. Can I add a TODO comment in both Expression::bindRValue and Expression::bindLValue for now, and let them be in a future pull request?

Reply:
Yeah, TODOs are fine.